### PR TITLE
DM-2469: Add URLs to PageImageComponents

### PR DESCRIPTION
**Pull request by emyl3**
### JIRA issue link
[DM-2469](https://agile6.atlassian.net/browse/DM-2469)

## Description - what does this code do?
This PR adds the ability to add a link to a `PageImageComponent`. The link can be an internal or external one.

## Testing done - how did you test it/steps on how can another person can test it 
1. Log in as an admin.
2. Go to `/admin/pages` and edit a page.
3. Add an image component without a URL.
4. Ensure the show page of the custom displays the image correctly and it has no URL.
5. Go to `/admin/pages` and edit the same page.
6. Add/Change the image component to include an internal URL.
7. Ensure the show page of the custom displays the image correctly and when you click it it leads to the internal page you designated.
8. Go to `/admin/pages` and edit the same page.
9. Add/Change an image component to include an external URL.
10. Ensure the show page of the custom displays the image correctly and when you click it it leads to the external page you designated.
11.  Go to `/admin/pages` and edit the same page.
12. Try different URLs (internal and external) that are malformed to ensure the page does not save.

## Acceptance criteria
- [X] Picture can be a hyperlink
- [X] Picture is a hyperlink to outside of marketplace page
- [X] Header photo needs to be hyperlink 

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by bradjohnson92008 — APPROVED** [review]
Great job Elisa - I was able to get through all the tests... tested several internal/external urls.  Malformed urls did not save, migration looks good and everything worked as expected!

File: spec/models/concerns/external_url_validator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    context 'given a URL with no domain extension' do
+      it 'should add an error message' do
+        component = PageImageComponent.new(page_image: @attachment, alt_text: 'alt image', url: 'https://www.google')
+        component.valid?
+        expect(component.errors[:url][0]).to eq("Not a valid external URL")
+      end
+    end
+
+    context 'given a URL with http' do
+      it 'should add an error message' do
```
[the remark this answers is not in the document]
Comment (emyl3) [reply spec/models/concerns/external_url_validator_spec.rb:30]: oops you are right! will fix that up :)

**Review by JoshingYou1 — APPROVED** [review]
Awesome work Elisa!! Everything functions as intended for me!

**Review by bradjohnson92008 — APPROVED** [review]
lgtm.  Thx Elisa

**Pull request merged**